pkg/aspect/accessLogsManager.go

```diff
@@ -32,98 +33,92 @@ type (
+
 	var templateStr string
-	switch logFormat {
+	switch cfg.Log.LogFormat {
 	case aconfig.COMMON:
-		attrNames = commonLogAttributes
 		templateStr = commonLogFormat
 	case aconfig.COMBINED:
-		attrNames = combinedLogAttributes
 		templateStr = combinedLogFormat
 	case aconfig.CUSTOM:
 		fallthrough
 	default:
-		templateStr = logCfg.CustomLogTemplate
-		attrNames = logCfg.Attributes
+		// Hack because user's can't give us descriptors yet. For now custom templates can be created by
+		// defining a "template" input. This is not documented anywhere but here.
+		templateStr = c.Aspect.Inputs["template"]
 	}
 
-	// should never result in error, as this should fail ValidateConfig()
-	if tmpl, err = template.New("accessLogsTemplate").Parse(templateStr); err != nil {
-		return nil, err
+	// TODO: when users can provide us with descriptors, this error can be removed due to validation
+	tmpl, err := template.New("accessLogsTemplate").Parse(templateStr)
+	if err != nil {
+		return nil, fmt.Errorf("log %s failed to parse template '%s' with err: %s", cfg.LogName, templateStr, err)
 	}
 
-	if aspect, err = a.(adapter.AccessLogsBuilder).NewAccessLogsAspect(env, c.Builder.Params.(adapter.AspectConfig)); err != nil {
-		return nil, err
+	asp, err := a.(adapter.AccessLogsBuilder).NewAccessLogsAspect(env, c.Builder.Params.(adapter.AspectConfig))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create aspect for log %s with err: %s", cfg.LogName, err)
 	}
 
 	return &accessLogsWrapper{
-		logName,
-		c.Aspect.GetInputs(),
-		aspect,
-		attrNames,
-		tmpl,
+		name:          cfg.LogName,
+		aspect:        asp,
+		labels:        cfg.Log.Labels,
+		template:      tmpl,
+		templateExprs: cfg.Log.TemplateExpressions,
 	}, nil
 }
 
 func (accessLogsManager) Kind() Kind { return AccessLogsKind }
 func (accessLogsManager) DefaultConfig() adapter.AspectConfig {
 	return &aconfig.AccessLogsParams{
-		LogName:   "access_log",
-		LogFormat: aconfig.COMMON,
-		// WARNING: we cannot set default attributes here, based on
-		// the params -> proto merge logic. These will override
-		// all other values. This should be mitigated by the move
-		// away from structpb-based config merging.
+		LogName: "access_log",
+		Log: &aconfig.AccessLogsParams_AccessLog{
+			LogFormat: aconfig.COMMON,
+		},
 	}
 }
 
 func (accessLogsManager) ValidateConfig(c adapter.AspectConfig) (ce *adapter.ConfigErrors) {
 	cfg := c.(*aconfig.AccessLogsParams)
-	if cfg.LogFormat != aconfig.CUSTOM {
-		return nil
+	if cfg.Log == nil {
+		ce = ce.Appendf("Log", "an AccessLog entry must be provided.")
+		return
 	}
-	tmplStr := cfg.CustomLogTemplate
-	if _, err := template.New("test").Parse(tmplStr); err != nil {
-		return ce.Appendf("CustomLogTemplate", "could not parse template string: %v", err)
+	if cfg.Log.LogFormat != aconfig.CUSTOM {
+		// If it's not custom we're using our own configs, so we're fine.
+		return nil
 	}
+	// TODO: validate custom templates when users can provide us with descriptors
```

Comment:
i'm confused by this change. why are we removing the validation check on custom templates? Have we lost the ability to specify custom templates?  If so, should we remove the option altogether now (OR have explicit warnings that CUSTOM is NOT supported)?